MapboxVision/src/main/java/com/mapbox/vision/sensors/SensorsManager.kt

```diff
@@ -77,32 +79,35 @@ internal class SensorsManager(application: Application) : SensorEventListener {
             return
         }
         synchronized(this) {
-            lastTimestamp = event.timestamp / 1000000
+            lastTimestamp = TimeUnit.NANOSECONDS.toMillis(event.timestamp)
             when (event.sensor.type) {
-                TYPE_ACCELEROMETER -> {
-                    userAccelerationWithGravity[0] = event.values[0]
-                    userAccelerationWithGravity[1] = event.values[1]
-                    userAccelerationWithGravity[2] = event.values[2]
+                Sensor.TYPE_ACCELEROMETER -> {
+                    event.values.copyInto(userAccelerationWithGravity, endIndex = userAccelerationWithGravity.size)
                 }
-                TYPE_MAGNETIC_FIELD -> {
-                    geomagnetic[0] = event.values[0]
-                    geomagnetic[1] = event.values[1]
-                    geomagnetic[2] = event.values[2]
+                Sensor.TYPE_MAGNETIC_FIELD -> {
+                    event.values.copyInto(geomagneticXyz, endIndex = geomagneticXyz.size)
```

Comment:
Cool extension!

Reply:
Btw, can we add `copyFrom` function that will use size of target and will allow to right it even shorter, eg.
`geomagneticXyz.copyFrom(event.values)` ?

Reply:
done